src/transaction/log_record.hpp

```diff
@@ -354,16 +354,108 @@ struct log_info_chkpt_trans
   LOG_LSA start_postpone_lsa;	/* Address of start postpone (if transaction was doing postpone during checkpoint) */
   char user_name[LOG_USERNAME_MAX];	/* Name of the client */
 
+
+  inline bool operator== (const log_info_chkpt_trans &ochkpt) const;
 };
 
+bool
+log_info_chkpt_trans::operator== (const log_info_chkpt_trans &ochkpt) const
+{
+  if (isloose_end != ochkpt.isloose_end)
+    {
+      return false;
+    }
+
+  if (trid != ochkpt.trid)
+    {
+      return false;
+    }
+
+  if (state != ochkpt.state)
+    {
+      return false;
+    }
+
+  if (head_lsa != ochkpt.head_lsa)
+    {
+      return false;
+    }
+
+  if (tail_lsa != ochkpt.tail_lsa)
+    {
+      return false;
+    }
+
+  if (undo_nxlsa != ochkpt.undo_nxlsa)
+    {
+      return false;
+    }
+
+  if (posp_nxlsa != ochkpt.posp_nxlsa)
+    {
+      return false;
+    }
+
+  if (savept_lsa != ochkpt.savept_lsa)
+    {
+      return false;
+    }
+
+  if (tail_topresult_lsa != ochkpt.tail_topresult_lsa)
+    {
+      return false;
+    }
+
+  if (start_postpone_lsa != ochkpt.start_postpone_lsa)
+    {
+      return false;
+    }
+
+  if (strlen (user_name) != strlen (ochkpt.user_name))
+    {
+      return false;
+    }
+
+  for (int i = 0; i < strlen (user_name); i++)
+    {
+      if (user_name[i] != ochkpt.user_name[i])
+	{
+	  return false;
+	}
+    }
```

Comment:
Use std::strcmp from <cstring>
```suggestion
  if (std::strcmp (user_name, ochkpt.user_name) != 0)
    {
      return false;
    }
```